internal/config/profile.go

```diff
@@ -641,25 +641,39 @@ func (p *Profile) Save() error {
 	return viper.WriteConfigAs(p.Filename())
 }
 
+// OldMongoCLIConfigHome retrieves configHome path based used by mongoCLI.
+// Deprecated: MongoCLI versions below v1.24.0 use this path.
+func OldMongoCLIConfigHome() (string, error) {
+	if home := os.Getenv("XDG_CONFIG_HOME"); home != "" {
+		return "", errors.New("not applicable")
+	}
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%s/.config", home), nil
+}
+
 // MongoCLIConfigHome retrieves configHome path based used by mongoCLI.
 func MongoCLIConfigHome() (string, error) {
 	if home := os.Getenv("XDG_CONFIG_HOME"); home != "" {
-		return home, nil
+		return fmt.Sprintf("%s/mongocli", home), nil
```

Comment:
I don't know if that was agreed already on a spec but a more OS compatible way for the .config folder is [os.UserConfigDir](https://pkg.go.dev/os#UserConfigDir). Currently our code always defaults to `os.UserHomeDir() / .config` but the above mentioned code would try `$HOME/Library/Application Support` for macos and `%AppData%` for Windows which is probably what our customers would have expected.

Not blocking just mentioning here since we are changing folders for the CLI this might by the right moment to rethink this folder.

Reply:
I didn’t know about this. Thanks for sharing it! I am in favor of this change as we are already changing the location of the config. Tagging @gssbzn to see if we all agree on this change.

Reply:
sounds great, to see if I understood -  we would replace the whole logic within `MongoCLIConfigHome() `and use `os.UserConfigDir()`, furthermore we would no longer use `.config`. Just simply create a mongocli folder under `UserConfigDir`. Adding to this PR sounds reasonable.

Reply:
I added the changes for this, LMK if we can proceed with this approach.